File: app/src/main/java/com/example/neon/deardiary/diaryquery/DiaryQueryPresenter.java
package com.example.neon.deardiary.diaryquery;


import com.example.neon.deardiary.data.Diary;
import com.example.neon.deardiary.data.source.DataSource;

import java.util.ArrayList;

class DiaryQueryPresenter implements DiaryQueryContract.Presenter {
    private DiaryQueryContract.View mView;
    private DataSource mDataSource;

    DiaryQueryPresenter(DiaryQueryContract.View view, DataSource dataSource) {
        mView = view;
        mView.setPresenter(this);
        mDataSource = dataSource;
    }

    @Override
    public void start() {

    }

    @Override
    public void diaryQueryByKeyword(String keyword) {
        mDataSource.queryByKeyword(keyword, new DataSource.LoadDiaryCallBack() {
            @Override
            public void onDiaryLoaded(ArrayList<Diary> diaryList) {
                mView.querySuccessfully(diaryList);
            }

            @Override
            public void onDataNotAvailable() {
                mView.queryFailed();
            }
        });

    }
}

File: app/src/main/java/com/example/neon/deardiary/diaryquery/DiaryQueryPresenter.kt
package com.example.neon.deardiary.diaryquery

import com.example.neon.deardiary.data.Diary
import com.example.neon.deardiary.data.DataSource

import java.util.ArrayList

internal class DiaryQueryPresenter(
        private val mView: DiaryQueryContract.View,
        private val mDataSource: DataSource?
) : DiaryQueryContract.Presenter {

    init {
        mView.setPresenter(this)
    }

    override fun diaryQueryByKeyword(keyword: String) {
        mDataSource?.queryByKeyword(keyword, object : DataSource.LoadDiariesCallBack {
            override fun onDiaryLoaded(diaryList: ArrayList<Diary>) {
                mView.querySuccessfully(diaryList)
            }

            override fun onDataNotAvailable() {
                mView.queryFailed()
            }
        })
    }
}
